### PR TITLE
Fixes for 'umaskCD and more'

### PR DESCRIPTION
**Pull request by eclare108213**
- [x] Short (1 sentence) summary of your PR: 
Draft fixes for @TillRasmussen 's PR
- [x] Developer(s): 
@TillRasmussen 
- [x] Suggest PR reviewers from list in the column to the right.
- [ ] Please copy the PR test results link or provide a summary of testing completed below.
    ENTER INFORMATION HERE 
- How much do the PR code changes differ from the unmodified code? 
    - [ ] bit for bit
    - [ ] different at roundoff level
    - [ ] more substantial 
- Does this PR create or have dependencies on Icepack or any other models?
    - [ ] Yes
    - [ ] No
- Does this PR add any new test cases?
    - [ ] Yes
    - [ ] No
- Is the documentation being updated? ("Documentation" includes information on the wiki or in the .rst files from doc/source/, which are used to create the online technical docs at https://readthedocs.org/projects/cice-consortium-cice/. A test build of the technical docs will be performed as part of the PR testing.)
    - [ ] Yes
    - [ ] No, does the documentation need to be updated at a later time?
        - [ ] Yes
        - [ ] No 
- [ ] Please provide any additional information or relevant details below:

### CODE REVIEW

File: cicecore/cicedynB/dynamics/ice_dyn_evp.F90 [hunk 1]
```diff
@@ -412,8 +412,8 @@ subroutine evp (dt)
                          indxti      (:,iblk), indxtj      (:,iblk), & 
                          indxni      (:,iblk), indxnj      (:,iblk), & 
                          aiN       (:,:,iblk), nmass     (:,:,iblk), & 
-                         nmassdti  (:,:,iblk), fcorN_blk  (:,:,iblk), & 
-                         nmask     (:,:,iblk),                       & 
+                         nmassdti  (:,:,iblk), fcorN_blk  (:,:,iblk),& 
+                         nmask     (:,:,iblk), nmask     (:,:,iblk), &
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_evp.F90:416]: maybe the second mask in this line could be made into an optional variable

File: cicecore/cicedynB/dynamics/ice_dyn_evp.F90 [hunk 2]
```diff
@@ -1517,19 +1540,19 @@ subroutine stress_U   (nx_block,   ny_block,  &
...
-!                                                 hm(i,j),         hm(i,j+1),      &
-!                                                 hm(i+1,j+1),     hm(i+1,j),      &
-!                                                 tarea(i,j),      tarea(i,j+1),   &
-!                                                 tarea(i+1,j+1),  tarea(i+1,j),   &
-!                                                 DeltaU                          )
+         zetax2U = c0
+         etax2U = c0
+         rep_prsU = c0
+         
+         call viscous_coeffs_and_rep_pressure_T2U (zetax2T(i  ,j  ), zetax2T(i  ,j+1), &
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_evp.F90:1547]: This is an example how I prefer to align code by each element (parentheses, i's, j's, commas, &'s).  The last line of this argument list still needs to be aligned.

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 [hunk 3]
```diff
@@ -1913,18 +1948,14 @@ subroutine viscous_coeffs_and_rep_pressure (strength,  tinyarea, &
...
-         tmpcalcsw = strength/max(Deltasw,tinyarea)
-         tmpcalcse = strength/max(Deltase,tinyarea)
-      else
-         tmpcalcne = strength/(Deltane + tinyarea)
-         tmpcalcnw = strength/(Deltanw + tinyarea)
-         tmpcalcsw = strength/(Deltasw + tinyarea)
-         tmpcalcse = strength/(Deltase + tinyarea)
-      endif
+         tmpcalcne = capping*(strength/max(Deltane,  tinyarea))+ &
+                     (1-capping)*strength/(Deltane + tinyarea)
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_shared.F90:1952]: need to use c1 instead of 1

File: cicecore/cicedynB/dynamics/ice_dyn_shared.F90 [hunk 4]
```diff
@@ -1986,23 +2016,63 @@ subroutine viscous_coeffs_and_rep_pressure_T (strength, tinyarea, &
...
+                                                        maskT_11,  maskT_10, &
+                                                        Tarea_00,   Tarea_01, &
+                                                        Tarea_11,   Tarea_10, &
+                                                        deltaU,               &
+                                                        zetax2U, etax2U,      &
+                                                        rep_prsU)
+                              
+
+      real (kind=dbl_kind), intent(in):: &
+        zetax2T_00,zetax2T_10,zetax2T_11,zetax2T_01, &
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_shared.F90:2042]: At some point, add comments at the end of each line to identify the variables (with units).

**Review by eclare108213 — COMMENTED** [review]
just adding a few comments on things I noticed

File: cicecore/cicedynB/dynamics/ice_dyn_evp.F90 [hunk 5]
```diff
@@ -1517,19 +1540,19 @@ subroutine stress_U   (nx_block,   ny_block,  &
...
+                                                   zetax2T(i+1,j+1), zetax2T(i+1,j  ), &
+                                                   etax2T (i  ,j  ), etax2T (i  ,j+1), &
+                                                   etax2T (i+1,j+1), etax2T (i+1,j  ), &
```
Comment (eclare108213) [inline cicecore/cicedynB/dynamics/ice_dyn_evp.F90:1550]: This is an example how I prefer to align code by each element (parentheses, i's, j's, commas, &'s).  The last line of this argument list still needs to be aligned.

File: cicecore/cicedynB/dynamics/ice_dyn_evp.F90 (hunk 1 above)
Comment (TillRasmussen) [reply cicecore/cicedynB/dynamics/ice_dyn_evp.F90:416]: I think that I can eliminate the second mask and differentiate the two mask in the call as it is done for dyn_prep2 on E and N grid

**Comment by apcraig** [comment]
Closed.  This PR was merged into #33.

**Pull request closed without merging**